### PR TITLE
fix add remove add

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -383,6 +383,11 @@ impl Peer {
         let data = try!(cmd.write_to_bytes());
         let change_peer = get_change_peer_cmd(&cmd).unwrap();
 
+        if change_peer.get_change_type() == ConfChangeType::AddNode &&
+           change_peer.get_peer().get_id() <= self.peer_id() {
+            return Err(box_err!("add node must allocate larger peer id than current!"));
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: the new id must > all ids in current region.

File: tests/raftstore/test_conf_change.rs [hunk 2]
```diff
@@ -33,15 +33,15 @@ fn test_simple_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
...
+    // Now region 1 only has peer (1, 1);
     let (key, value) = (b"a1", b"v1");
 
     cluster.must_put(key, value);
     assert_eq!(cluster.get(key), Some(value.to_vec()));
 
     let engine_2 = cluster.get_engine(2);
     must_get_none(&engine_2, b"a1");
-    // add peer (2,2,2) to region 1.
+    // add peer (2,2) to region 1.
```
Comment (siddontang) [inline tests/raftstore/test_conf_change.rs]: insert a space. (2,2) -> (2, 2)

**Comment by siddontang** [comment]
Please merge master and update the partition API first.

**Comment by tiancaiamao** [comment]
after refacting pd https://github.com/pingcap/tikv/pull/588 , we don't meet a remove peer situations currently, so this test case is outdated.

**Pull request closed without merging**